src/state/state-manager.js

```diff
@@ -111,9 +112,18 @@ export default class StateManager {
         this._dispatchEvent();
       },
       [Html5EventType.WAITING]: () => {
-        this._updateState(StateType.BUFFERING);
-        this._lastWaitingTime = this._player.currentTime;
-        this._dispatchEvent();
+        if (this._player.seeking) {
+          this._updateState(StateType.SEEKING);
+          this._dispatchEvent();
+          //if waiting and seeking took more than threshold, it's buffering
+          setTimeout(() => {
+            if (StateType.SEEKING === this._curState.type) {
+              this._waiting();
+            }
+          }, SEEKING_THRESHOLD);
```

Comment:
I think it is better to add spinner in UI to account for seeking then to manage by timeout

Reply:
another option is to add reason to event,
for example `{reason: seek}` and then in UI we don't need to change a thing and in analytics we can check reason and not trigger buffer under run.